cupy/lib/_routines_poly.py

```diff
@@ -138,28 +139,19 @@ def polyval(p, x):
     out = p[::-1] * cupy.power(val, cupy.arange(p.size))
     out = out.sum(axis=1)
     dtype = cupy.result_type(p, val)
-    # For case: when p is of shape (0,) and x is (), output is
-    #  of single valued array to match NumPy's behavior
-    if isinstance(x, cupy.ndarray) and x.ndim == 0 and p.size == 0:
-        return out[0].astype(dtype, copy=False)
-    # to match the shape of NumPy's results
     if cupy.isscalar(x) or x.ndim == 0:
         return out.astype(dtype, copy=False).reshape()
-    # to match the dtype of NumPy's results
     if p.dtype == numpy.complex128 and val.dtype in [
        numpy.float16, numpy.float32, numpy.complex64]:
         return out.astype(numpy.complex64, copy=False)
```

Comment:
Not needed.

Reply:
test_polyval_diff_types_array_array fails